### DISCUSSION TITLE
Remove validation verifyingContract on eth_signTypedData_v4

### DISCUSSION THREAD

**Discussion by Grinion** [phantom/docs#222] [Help & Support] [answered]
### Summary

Hello,

I am currently in the process of integrating a project developed on the Injective blockchain with the Phantom wallet.

I have encountered an issue with the eth_signTypedData_v4 method. I need to sign an EIP-712 message containing 
'cosmos' in the verifyingContract field. When I try to do it I get an error, however, everything functions correctly when I use Ethereum address as verifyingContract

### Steps to Reproduce

```
let msg = {
    "types": {
        "EIP712Domain": [
            {
                "name": "name",
                "type": "string"
            },
            {
                "name": "version",
                "type": "string"
            },
            {
                "name": "chainId",
                "type": "uint256"
            },
            {
                "name": "verifyingContract",
                "type": "address"
            }
        ],
        "Mail": [
            {
                "name": "contents",
                "type": "string"
            }
        ]
    },
    "primaryType": "Mail",
    "domain": {
        "name": "Ether Mail",
        "version": "1",
        "chainId": 1,
        "verifyingContract": "cosmos"
    },
    "message": {
        "contents": "Hello, Bob!"
    }
}


window.phantom.ethereum.request({
                method: 'eth_signTypedData_v4',
                params: [window.phantom.ethereum.selectedAddress, JSON.stringify(msg)],
})  // here I get error



msg.domain.verifyingContract = "0x26c4D7DC8d07E9863277b768f923f5BBDEE66BC6"

window.phantom.ethereum.request({
                method: 'eth_signTypedData_v4',
                params: [window.phantom.ethereum.selectedAddress, JSON.stringify(msg)],
}) // working now


```

### Phantom Version

24.5.0

### Is there an existing discussion for this?

- [X] I have searched the existing discussions

**Comment by adamdelphantom** [comment]
Hello @Grinion,

Thank you for writing in! Currently we do not support connecting Phantom to the Injective network. We can't guarantee compatibility currently or in the future. Please consider submitting or upvoting a related feature request to support Injective, [here](https://phantom.canny.io).

**Reply by Grinion** [reply]
Hello @adamdelphantom   
I'm not asking you to support the Injective blockchain. Now I have a problem with Injective right now - but other projects may have the same problem with other projects. 

I think you could consider disabling the validation of this field as it is done in metamask and trustwallet.  It's not necessary for anything

**Reply by Grinion** [reply]
Hello @adamdelphantom 

I have created a feature request but I think it's more a bug than a feature. Please disable validation it will take only 1 minute and it will help the project to create the integration. Thank you
https://phantom.canny.io/feature-requests/p/remove-validation-verifyingcontract-on-eth-signtypeddata-v4

**Reply by adamdelphantom** [reply, marked as answer]
@Grinion Thank you for submitting the feature request.

Our transaction and message signing scanning is a top concern for Phantom. Any redaction takes significant resources to ensure it is the right decision for the protection of Phantom users. I'm afraid doing so to unlock an unsupported use case is not something we can prioritize in the foreseeable future. I hope you understand. And I do appreciate your desire to user Phantom. Hopefully we can meet support your preferred network in the future.
